src/processor/operator/persistent/reader/csv/driver.cpp

```diff
@@ -522,6 +599,81 @@ void copyStringToVector(ValueVector* vector, uint64_t rowToAdd, std::string_view
     }
 }
 
+template<>
+void copyStringToVector(ValueVector* vector, uint64_t rowToAdd, std::string_view strVal,
```

Comment:
Every copyStringToVector specialization has this common code block:
```
if (strVal.empty() || isNull(strVal)) {
        vector->setNull(rowToAdd, true /* isNull */);
        return;
    } else {
        vector->setNull(rowToAdd, false /* isNull */);
    }
```
The only difference is the logic to cast and set value in the vector. So you only need to template the cast and set value logic.